src/blueprint/upstream.rs

```diff
@@ -32,7 +32,7 @@ pub struct Upstream {
 impl Upstream {
     pub fn is_batching_enabled(&self) -> bool {
         if let Some(batch) = self.batch.as_ref() {
-            batch.delay >= 1 || batch.max_size >= 1
+            batch.delay >= 1 || batch.max_size.unwrap_or(100) >= 1
```

Comment:
If `max_size` is not set, it will default to 100, concluding that batching is enabled, when it's not.